xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -1516,24 +1516,22 @@ public InputStream getResourceAsStream(String s) throws MalformedURLException
 
     public String getResourceContent(String name) throws IOException
     {
-        InputStream is = null;
+        InputStream temp = null;
         if (getEngineContext() != null) {
 
             try {
-                is = getResourceAsStream(name);
+                temp = getResourceAsStream(name);
             } catch (Exception e) {
             }
         }
 
-        if (is == null) {
+        if (temp == null) {
             // Resources should always be encoded as UTF-8, to reduce the dependency on the system encoding
             return FileUtils.readFileToString(new File(name), DEFAULT_ENCODING);
         }
 
-        try {
+        try (InputStream is = temp) {
```

Comment:
Same comment as in the other pull request, this whole block should be moved and merged with the content of the `if (getEngineContext() != null) {` block to make things a lot cleaner.